src/Tracking/DisasterTrackerApp.BL/Implementation/WarningService.cs

```diff
@@ -47,7 +50,23 @@ where d.Geometry.IsWithinDistance((Geometry)c.Coordiantes,MaxRadius)
                     ))
                 .SelectMany(e=>e);
         }
-
+        public IObservable<WarningDto> GetStatisticsWarningEvents(WarningRequest warningRequest,
+            CancellationToken cancellationToken = default)
+        {
+            return Observable.FromAsync(async () =>
+                from c in await _calendarRepository.GetCalendarEventsFilteredWithUserId(warningRequest.UserId,
+                    BuildExpression(warningRequest),
+                    cancellationToken)
+                from d in _disasterEventRepository.GetAllDisasterEvents()
+                where d.Geometry.Distance(c.Coordiantes) <= MaxRadius
+                select new WarningDto(c.Id,
+                    d.Id,
+                    $"Warning. Disaster can occur near your event in place {c.Location}",
+                          c.EndTs,
+                      c.StartedTs
```

Comment:
is it possible to rewrite it to the repository level? 
you can add logs to see query https://stackoverflow.com/questions/56646186/how-to-enable-logging-in-ef-core-3
when it is Enumerable - means is already executed on db side and returned a response to memory
when it is IQuerable - it still can use the efficiency of db computing.